backtracking/wildcard_matching.cpp

```diff
@@ -0,0 +1,155 @@
+/**
+ * @file
+ * @brief Implementation of the [Wildcard
+ * Matching](https://www.geeksforgeeks.org/wildcard-pattern-matching/) problem.
+ * @details
+ * Given a matching string and a pattern, implement wildcard pattern
+ * matching with support for `?` and `*`. `?` matches any single character.
+ * `*` matches any sequence of characters (including the empty sequence).
+ * The matching should cover the entire matching string (not partial). The task
+ * is to determine if the pattern matches with the matching string
+ * @author [Swastika Gupta](https://github.com/Swastyy)
+ */
+
+#include <cassert>   /// for assert
+#include <iostream>  /// for IO operations
+#include <vector>    /// for std::vector
+
+/**
+ * @namespace backtracking
+ * @brief Backtracking algorithms
+ */
+namespace backtracking {
+/**
+ * @namespace wildcard_matching
+ * @brief Functions for the [Wildcard
+ * Matching](https://www.geeksforgeeks.org/wildcard-pattern-matching/) problem.
+ */
+namespace wildcard_matching {
+/**
+ * @brief The main function implements if pattern can be matched with given
+ * string
+ * @param s is the given matching string
+ * @param p is the given pattern
+ * @param pos1 is the starting index
+ * @param pos2 is the last index
+ * @returns 1 if pattern matches with matching string otherwise 0
+ */
+std::vector<std::vector<int64_t>> dpTable(1000, std::vector<int64_t>(1000, -1));
+uint32_t wildcard_matching(std::string s, std::string p, uint32_t pos1,
```

Comment:
```suggestion
std::vector<std::vector<int64_t>> dpTable(1000, std::vector<int64_t>(1000, -1));
bool wildcard_matching(std::string s, std::string p, uint32_t pos1,
```
If this function is only returning 1/0 instead of returning `uint32_t` we can return true or false which i feel will be more efficient. What do you think @ayaankhan98 @Swastyy @Panquesito7 ?

Reply:
For me, that sounds good. Just make sure the self-tests still pass.

Reply:
Some very little change in space maybe 🤔